src/zenml/logging/step_logging.py

```diff
@@ -529,70 +670,8 @@ def __exit__(
         except (OSError, IOError) as e:
             logger.warning(f"Step logs roll-up failed: {e}")
 
-    def _wrap_write(self, method: Callable[..., Any]) -> Callable[..., Any]:
-        """Wrapper function that utilizes the storage object to store logs.
-
-        Args:
-            method: the original write method
-
-        Returns:
-            the wrapped write method.
-        """
-
-        def wrapped_write(*args: Any, **kwargs: Any) -> Any:
-            step_names_disabled = (
-                handle_bool_env_var(
-                    ENV_ZENML_DISABLE_STEP_NAMES_IN_LOGS, default=False
-                )
-                or not self.prepend_step_name
-            )
-
-            if step_names_disabled:
-                output = method(*args, **kwargs)
-            else:
-                message = args[0]
-                # Try to get step context if not available yet
-                step_context = None
-                try:
-                    step_context = get_step_context()
-                except Exception:
-                    pass
-
-                if step_context and args[0] not in ["\n", ""]:
-                    # For progress bar updates (with \r), inject the step name after the \r
-                    if "\r" in message:
-                        message = message.replace(
-                            "\r", f"\r[{step_context.step_name}] "
-                        )
-                    else:
-                        message = f"[{step_context.step_name}] {message}"
-
-                output = method(message, *args[1:], **kwargs)
-
-            # Save the original message without step name prefix to storage
-            if args:
-                self.storage.write(args[0])
-
-            return output
-
-        return wrapped_write
-
-    def _wrap_flush(self, method: Callable[..., Any]) -> Callable[..., Any]:
-        """Wrapper function that flushes the buffer of the storage object.
-
-        Args:
-            method: the original flush method
-
-        Returns:
-            the wrapped flush method.
-        """
-
-        def wrapped_flush(*args: Any, **kwargs: Any) -> Any:
-            output = method(*args, **kwargs)
-            self.storage.save_to_file()
-            return output
-
-        return wrapped_flush
+        # Reset the step names context to default
+        step_names_in_console.set(False)
```

Comment:
What if another context is registered that still wants to have step names in the console?